src/Mobile.BuildTools/Generators/ReleaseNotesGenerator.cs

```diff
@@ -22,7 +22,13 @@ protected override void ExecuteInternal()
             if (releaseNotesOptions.Disable) return;
 
             var currentHash = GetGitHash();
-            var generatedMarkerFilePath = Path.Combine(Build.SolutionDirectory, ".vs", $"generated-releasenotes.{currentHash}");
+            var generatedMarkerDirectory = Path.Combine(Build.SolutionDirectory, ".vs");
+
+            if (!Directory.Exists(generatedMarkerDirectory))
+                Directory.CreateDirectory(generatedMarkerDirectory);
+
+            var generatedMarkerFilePath = Path.Combine(generatedMarkerDirectory, $"generated-releasenotes.{currentHash}");
+

```

Comment:
can you convert this to a FileInfo

```cs
var generatedMarkerFilePath = new FileInfo(Path.Combine(Build.SolutionDirectory, ".vs", $"generated-releasenotes.{currentHash}"));

if(!generatedMarkerFilePath.Directory.Exists)
    generatedMarkerFilePath.Directory.Create();
```

Reply:
@dansiegel Can do. Same for the actual release notes too - like below? I did just copy the previous pattern as thought keeping them aligned made for better reading. Your decision though :)

```
var generatedMarkerFilePath = new FileInfo(Path.Combine(Build.SolutionDirectory, ".vs", $"generated-releasenotes.{currentHash}"));

if (!generatedMarkerFilePath.Directory.Exists)
	generatedMarkerFilePath.Directory.Create();

var releaseNotesFilePath = new FileInfo(Path.Combine(releaseNotesOptions.CreateInRoot ? Build.SolutionDirectory : Build.ProjectDirectory, releaseNotesOptions.FileName));

if (!releaseNotesFilePath.Directory.Exists)
	releaseNotesFilePath.Directory.Create();

if (releaseNotesFilePath.Exists && generatedMarkerFilePath.Exists)
{
	Log.LogMessage("Current Release Notes file has already been generated.");
	return;
}
```